# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PWindowGlobal.ipdl
# Commit: 750e0796c1ad
# Full Hash: 750e0796c1ad0a93d26428fefe4419af324371a4
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-07 21:34:02
# Regressor Bug: 1839918
# File Overlap Count: 3
# Description:
#   Bug 1839918 - Create SetCookies variant for document.cookie on PWindowGlobal. r=bvandersloot,cookie-reviewers,valentin
#   
#   Using PWindowGlobal for SetCookies calls from content to parent process means we can get
#   the BrowsingContext the cookies are set for. The current implementation in PCookieService
#   drops this relationship.
# ==============================================================================

diff -r 3949d9b400d4 -r 750e0796c1ad dom/ipc/PWindowGlobal.ipdl
--- a/dom/ipc/PWindowGlobal.ipdl	Thu Sep 07 08:25:55 2023 +0000
+++ b/dom/ipc/PWindowGlobal.ipdl	Thu Sep 07 08:25:55 2023 +0000
@@ -212,6 +212,13 @@
 
   async HasStorageAccessPermission() returns(bool granted);
 
+
+  async SetCookies(nsCString baseDomain,
+                                      OriginAttributes attrs,
+                                      nullable nsIURI host,
+                                      bool fromHttp,
+                                      CookieStruct[] cookies);
+
 child:
   async NotifyPermissionChange(nsCString type);
 };